src/app/WriteHandler.cpp

```diff
@@ -73,7 +75,10 @@ Status WriteHandler::OnWriteRequest(Messaging::ExchangeContext * apExchangeConte
         }
     }
 
-    Shutdown();
+    if (!(status == Status::Success && mHasMoreChunks))
```

Comment:
OK, so if mHasMoreChunks is set but we never get another chunk... we just leak?  That does not seem good.

Reply:
:heavy_check_mark:  Code updated,